legacy/firmware/ethereum.c

```diff
@@ -313,30 +315,31 @@ static void send_signature(void) {
  * using standard ethereum units.
  * The buffer must be at least 25 bytes.
  */
-static void ethereumFormatAmount(const bignum256 *amnt, const TokenType *token,
-                                 char *buf, int buflen) {
+static void ethereumFormatAmount(const bignum256 *amnt,
+                                 const EthereumTokenInfo *token, char *buf,
+                                 int buflen) {
   bignum256 bn1e9 = {0};
   bn_read_uint32(1000000000, &bn1e9);
   const char *suffix = NULL;
   int decimals = 18;
-  if (token == UnknownToken) {
+  if (is_UnknownToken(token)) {
```

Comment:
istm we can drop this whole branch and the function `is_UnknownToken` too, because our `UnknownToken` has the right symbol + decimals?

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562